TAO/TAO_IDL/driver/drv_preproc.cpp

```diff
@@ -1188,15 +1189,47 @@ DRV_pre_proc (const char *myfile)
 
   // Rename temporary files so that they have extensions accepted
   // by the preprocessor.
+  const char *call_failed = nullptr;
+  bool bail = false;
   FILE * const file = ACE_OS::fopen (myfile, "r");
-
   if (file == nullptr)
+    call_failed = "";
+  else
+    {
+      // Check to see that the file is a normal file. If we don't and the file
+      // is a directory, then the copy would be blank and any error message
+      // later wouldn't be accurate.
+      const ACE_HANDLE file_desc = ACE_OS::fileno (file);
+      if (file_desc == ACE_INVALID_HANDLE)
+        call_failed = "(fileno) ";
+      else
+        {
+          ACE_stat file_stat;
+          if (ACE_OS::fstat (file_desc, &file_stat) == -1)
```

Comment:
ACE_OS has a `stat` function which does both of these in one operation. Any reason not to use it?

Reply:
Lack of knowledge of said function.

Reply:
I don't see a `stat` function that takes a `FILE*`, only the one that takes the path.

Reply:
I thought it would start with `myfile` since that's the input to the function.   I guess I should have picked a different line to write the comment on.